variables.tf

```diff
@@ -57,6 +57,12 @@ variable "lifecycle_rule" {
   description = "List of maps containing lifecycle management configuration settings"
 }
 
+variable "object_lock_configuration" {
+  type        = map
+  default     = {}
+  description = "Bucket object lock configuration settings. Needs to be set upon creation of the bucket. Valid keys: 'mode' (required) and one or both of 'years' or 'days'"
+}
```

Comment:
We already have one `map` variable which makes the configuration less clear. So lets not introduce yet another one, but instead define this one as a proper object. Especially since this object only needs 3 fields, there isn’t much reason not to use an object IMO.

Reply:
Object doesn't support optional arguments

Reply:
I know, so then I would prefer 3 separate variables: `lock_mode`, `lock_years` and `lock_days`. You can then also add some validation in the vars if needed, see [here](https://www.terraform.io/docs/configuration/variables.html#custom-validation-rules) for reference.